modules/local_contacts/modules.php

```diff
@@ -65,6 +65,92 @@ public function process() {
     }
 }
 
+/**
+ * @subpackage local_contacts/handler
+ */
+class Hm_Handler_process_import_contact extends Hm_Handler_Module {
+    public function process() {
+        list($success, $form) = $this->process_form(array('contact_source', 'import_contact'));
+        if ($success && $form['contact_source'] == 'csv') {
+            $file = $this->request->files['contact_csv'];
+            $csv = fopen($file['tmp_name'], 'r');
+            if ($csv) {
+                $contacts = $this->get('contact_store');
+                $header = fgetcsv($csv);
+                $expectedHeader = array('display_name', 'email_address', 'phone_number');
+
+                if ($header !== $expectedHeader) {
+                    fclose($csv);
+                    Hm_Msgs::add('ERRInvalid CSV file, please use a valid header: '.implode(', ', $expectedHeader));
+                    return;
+                }
+
+                $contact_list = $this->user_config->get('contacts', array());
+                $contact_list = array_map(function($v) { $v['type'] = 'local'; return $v; }, $contact_list);
```

Comment:
There was a recent change for global unique IDs that affected this code. The contacts are automatically fetched from user config now in contacts module Hm_Handler_load_contacts. Please rebase latest master and update your code accordingly. Basically, you don't need to map the list of contacts here - just get the contact store and getAll or execute another method on the repository.